### PR TITLE
Make FINESSE buildable with `pyinstaller`

### PR DESCRIPTION
**Pull request by alexdewar**
This allows for building FINESSE by running `pyinstaller FINESSE.spec` (the output is put in `dist/`). Tested on Windows and Linux and it runs without a hitch!

I decided to enable the option to build everything into a single (compressed) executable, so that when we get to publishing these files on GitHub somewhere (#12), they're easier to work with (i.e. you don't have to extract a zip file anything).

I ended up having to use a stub entry point again :cry:. @dalonsoa Did you figure that out with iPANACEA at some point?

### CODE REVIEW

**Comment by dalonsoa** [comment]
That was solved in this PR, but it was never merged... https://github.com/ImperialCollegeLondon/iPANACEA

In summary, you can directly used `__main__.py`

**Review by dalonsoa — CHANGES_REQUESTED** [review]
Use `__main__.py`, not a stub.

**Comment by alexdewar** [comment]
I tried using `finesse/__main__.py` as an entry point instead of the stub, but I get the following error:

```
ImportError: attempted relative import with no known parent package
```

I think the problem is that in `finesse/__main__.py` there are some relative imports (cf. iPANACEA's `__main__.py`), so Python is throwing a wobbly:

```py
from . import config
from .gui.main_window import MainWindow
```

Any ideas?

File: stub.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""Stub script for pyinstaller."""
+from finesse.__main__ import main
```
Comment (dalonsoa) [inline stub.py:2]: I think that the options we have are either use absolute imports in `__main__.py` or, indeed, use a stub. Nothing against it.

I would re-organise a bit some things, though, as importing atom `__main__` just does not look correct:

1. Move the function that configures the logger to a `logger.py` file
2. Move the `main` function to `__init__.py`, importing there from logger, main window, etc.
3. Make `__main__.py` and `stub.py` identical, just one doing `from . import main` and the other `from finesse import main`.

Does this make sense?

**Review by dalonsoa — CHANGES_REQUESTED** [review]

File: stub.py (hunk 1 above)
Comment (alexdewar) [reply stub.py:2]: Yup, sounds good!

**Review by dalonsoa — APPROVED** [review]
Keeping aside an issue with `poetry.lock` (as usual), all looks good!

**Pull request merged**